app/interactors/content/update_interactor.rb

```diff
@@ -58,7 +58,9 @@ def update_preview
   end
 
   def change_note_params
-    params.require(:revision).permit(:update_type, :change_note)
+    return {} if edition.first?
+
+    { update_type: params[:update_type], change_note: params[:change_note] }.compact
```

Comment:
Do we need the compact here? Since we don't do this for fields I wonder if it's an inconsistency we need or not.

Reply:
Yes, this was a surprise. The problem is that the presence of the key will modify the revision, as the value being `nil` is significant in general e.g. for lead images. So we need to remove redundant keys - the tests helpfully fail otherwise.

Reply:
Ah bummer - nils and empty strings are so annoying. I wonder if we should just have loads of our string fields be not null with defaults of empty string to avoid these challenges.

Reply:
This sounds interesting. Could you explain more what you mean here?

Reply:
Sure, no worries.

So in the scenario where we have a table like: 

```
create_table "our_table", force: :cascade do |t|
  t.string "our_field"
end
```

We have the situation where our_field can have a value of an empty string or null which means we have to worry about two different empty states. Whereas in some ways life might be easier if we have a default of empty string:

```
create_table "our_table", force: :cascade do |t|
  t.string "our_field", null: false, default: ''
end
```

Then there is only one empty state. 

We've hit a couple of issues like this with migration where Publishing API has an empty string whereas we have a nil value in our DB.

Loosely ties into an article I saw yesterday actually: https://mitchum.blog/null-values-in-sql-queries/

Reply:
Ahhh, I see. I thought you might be referring to some HTML form behaviour that would transform the nulls into empty strings. Turns out we can expect the parameters to be non-null in practice, so I've modified this commit to avoid `compact` 😸.